iam/src/providers/githubOrg.ts

```diff
@@ -0,0 +1,124 @@
+// ----- Types
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+import type { Provider, ProviderOptions } from "../types";
+import { requestAccessToken } from "./github";
+import axios from "axios";
+
+export type GithubTokenResponse = {
+  access_token: string;
+};
+
+export type GithubFindMyUserResponse = {
+  id?: string;
+  login?: string;
+  type?: string;
+};
+
+type GHUserRequestPayload = RequestPayload & {
+  org?: string;
+};
+
+type GithubMyOrg = {
+  providedOrg: string;
+  matchingOrg: string;
+};
+
+type GHVerification = {
+  validOrg: GithubMyOrg;
+  handle: string;
+};
+
+// Export a Github Provider to carry out OAuth and return a record object
+export class GithubOrgProvider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "GithubOrg";
+
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the Github user is a memeber of the provided organization
+  async verify(payload: GHUserRequestPayload): Promise<VerifiedPayload> {
+    let valid = false,
+      ghVerification: GHVerification;
+
+    try {
+      ghVerification = await verifyGithub(payload.proofs.code, payload.org);
+    } catch (e) {
+      return { valid: false };
+    } finally {
+      const validOrg = ghVerification?.validOrg;
+      valid = validOrg && validOrg.matchingOrg === validOrg.providedOrg;
+    }
+
+    return {
+      valid: valid,
+      record: {
+        org: ghVerification.validOrg.matchingOrg,
+        handle: ghVerification.handle,
```

Comment:
Sorry to do this to you @schultztimothy but this should really be `ghVerification.id`, the user can change their handle but not their ID (I'd forgotten about this, @aminah-io reminded me yesterday).

```suggestion
        id: ghVerification.id,
```